cmd/export-distro/main.go

```diff
@@ -121,9 +121,12 @@ func loadConfig() (distro.Config, config) {
 		ConsulPort: defConsulPort,
 		Hostname:   env(envDistroHost, defHostname),
 	}
-	hostname, err := os.Hostname()
-	if err == nil {
-		cfg.Hostname = hostname
+	// if using default hostname, try to use os.hostname
+	if cfg.Hostname == defHostname{
```

Comment:
This is not how the algorithm is supposed to work.

Hostname is set to default here: https://github.com/edgexfoundry/edgex-go/blob/master/cmd/export-distro/main.go#L122

Actually, this whole part: https://github.com/edgexfoundry/edgex-go/blob/master/cmd/export-distro/main.go#L124-L128 does not make sense and should be removed.

`cfg.Hostname` should be either set to default, or from an ENV variable. Service should not try to guess `hostname` from underlying OS by using `os` module.

While you are at it, you can remove this one also: https://github.com/edgexfoundry/edgex-go/blob/master/cmd/export-client/main.go#L150-L153